### PR TITLE
(DOCSP-37831) Update Private Endpoints documentation

### PR DESCRIPTION
**Pull request by krollins-mdb**
## Pull Request Info

Jira ticket: https://jira.mongodb.org/browse/DOCSP-37831

- [Use a VPC Private Endpoint](https://preview-mongodbkrollinsmdb.gatsbyjs.io/atlas-app-services/DOCSP-37831/security/private-endpoints/):
  - updated the API sections to be fully programmatic
  - updated procedures to account for new UI flow
  - Added new information about Legacy, General, and Sync Endpoints
  - Added a "Before You Begin" section
 
This PR shouldn't be merged until right before the 04/17 BAAS release. There's a related Jira ticket to handle updating the Admin API spec: https://jira.mongodb.org/browse/DOCSP-38290

### Reminder Checklist

Before merging your PR, make sure to check a few things.

- [x] Did you tag pages appropriately?
  - genre
  - programming_language
  - meta.keywords
  - meta.description
- [x] Describe your PR's changes in the Release Notes section
- [x] Create a Jira ticket for related docs-realm work, if any

### Release Notes

- **Security**
  - Use a VPC Private Endpoint: Rewrite page to account for new Endpoint creation process with Legacy, General, and Sync Endpoints.

### Review Guidelines

[REVIEWING.md](https://github.com/mongodb/docs-app-services/blob/master/REVIEWING.md)

### CODE REVIEW

File: source/security/private-endpoints.txt [hunk 1]
```diff
@@ -215,20 +290,18 @@ connection using either the App Services UI or the Admin API.
                  -d '{
                    "cloud_provider_region": "<your-cloud-deployment-region>",
                    "cloud_provider_endpoint_id": "<your-vpc-endpoint-id>"
+                   "endpoint_type":"sync"
                  }'
+            
+            #. You must also create a General Private Endpoint by calling
+               the ``Create a VPC Private Endpoint`` Admin API endpoint again.
+               This time, set the ``endpoint_type`` to ``"general``.
```
Comment (krollins-mdb) [inline source/security/private-endpoints.txt:298]: For technical reviewer: is this true? Or does hitting that Admin API endpoint automatically create the General Endpoint too?

File: source/security/private-endpoints.txt [hunk 2]
```diff
@@ -215,20 +290,18 @@ connection using either the App Services UI or the Admin API.
...
-            .. code-block:: sh
+Migrate a Legacy Endpoint
+~~~~~~~~~~~~~~~~~~~~~~~~~
 
-               aws ec2 modify-vpc-endpoint \
-                 --region <your-aws-deployment-region> \
-                 --vpc-endpoint-id <your-vpc-endpoint-id> \
-                 --private-dns-enabled
+To migrate a Legacy Endpoint to a Sync or General endpoint, you must create
+new endpoints that use your existing VPC endpoints.
```
Comment (krollins-mdb) [inline source/security/private-endpoints.txt:304]: > that use your existing VPC endpoints

Is this true? Do we want people using their existing VPC endpoints? Or should we encourage them to create new VPC endpoints too?

Any other considerations for migrating?

File: source/security/private-endpoints.txt [hunk 3]
```diff
@@ -252,3 +325,49 @@ You can toggle this setting in three ways:
 - In the Admin API, call the :admin-api-endpoint:`Toggle Non-VPC Requests
   <operation/allNonVpcClientRequests>` with a boolean value (``true`` to remove
   the default restriction, or ``false`` to reinstate it.)
+
+Definitions
```
Comment (krollins-mdb) [inline source/security/private-endpoints.txt:329]: I figured it would be good to have a reference of sorts that explains the differences between these things. I haven't quite gotten to writing those definitions. If you have suggestions, I'd love to hear them!

File: source/security/private-endpoints.txt [hunk 4]
```diff
@@ -252,3 +325,49 @@ You can toggle this setting in three ways:
...
+-----------
+
+There are three kinds of Private Endpoints:
+
+- Legacy: Endpoints that use ``realm.mogodb.com``
+- General: TODO: Add definition
+- Sync: TODO: Add definition
```
Comment (melodeeli98) [inline source/security/private-endpoints.txt:336]:
```suggestion
- Legacy: Endpoints that use the legacy ``realm.mongodb.com`` domain. Apps with legacy endpoints must migrate their endpoints to use the new domain.
- General: Endpoints that use ``services.cloud.mongodb.com`` for Client HTTP requests.
- Sync: Endpoints that use ``services.cloud.mongodb.com`` for Sync WebSocket connections.
```

File: source/security/private-endpoints.txt [hunk 5]
```diff
@@ -4,6 +4,9 @@
 Use a VPC Private Endpoint (Preview)
 ====================================
 
+.. meta:: 
+   :description: Learn how to set up a VPC Private Endpoint for your Atlas Cluster.
```
[the remark this answers is not in the document]
Comment (krollins-mdb) [reply source/security/private-endpoints.txt:8]: Ah, yep. Thanks for the correction. This is meta data for the page and could show up in search results (Google, docs search, etc).

File: source/security/private-endpoints.txt (hunk 4 above)
Comment (krollins-mdb) [reply source/security/private-endpoints.txt:336]: Thank you for filling in the gaps here. 😄

File: source/security/private-endpoints.txt [hunk 6]
```diff
@@ -17,36 +20,16 @@ AWS PrivateLink. To learn more, see `What is AWS PrivateLink?
...
 in the AWS documentation.
 
-Private Endpoint Limitations
-----------------------------
+Before You Begin
+----------------
 
-- You cannot use Private Endpoints with globally deployed Apps or local
-  Apps deployed to Azure or GCP.
+You should set up a Private Hosted Zone. To do so, use ``ModifyVpcAttribute``
```
[the remark this answers is not in the document]
Comment (krollins-mdb) [reply source/security/private-endpoints.txt:26]: Yep! Can do.

File: source/security/private-endpoints.txt [hunk 7]
```diff
@@ -68,28 +51,38 @@ following information:
```
[the remark this answers is not in the document]
Comment (krollins-mdb) [reply source/security/private-endpoints.txt:51]: Nice catch. And I'll note that legacy endpoints can't be created in relevant places.

File: source/security/private-endpoints.txt [hunk 8]
```diff
@@ -100,111 +93,193 @@ connection using either the App Services UI or the Admin API.
...
 
-            #. Enter your **Subnet IDs** as a comma-separated list in the input box.
+      .. procedure::
 
-            Once you've entered this information, the fully constructed
-            AWS CLI command is displayed in the UI. Copy the command and
-            run it in your terminal. It should resemble the following:
+         .. step:: Create an Interface Endpoint in Your VPC
 
+            a. In the App Services UI, navigate to :guilabel:`App Settings` and
```
[the remark this answers is not in the document]
Comment (krollins-mdb) [reply source/security/private-endpoints.txt:141]: Thank you for pointing this out! Makes sense to have this be totally programmatic.

File: source/security/private-endpoints.txt [hunk 9]
```diff
@@ -100,111 +93,193 @@ connection using either the App Services UI or the Admin API.
...
 
-         .. step:: Modify Your VPC Private DNS
+               curl https://services.cloud.mongodb.com/api/admin/v3.0/groups/{groupId}/apps/{appId}/security/private_endpoints \
+                 -X POST \
+                 -H "Authorization: Bearer <accessToken>" \
+                 -H "Content-Type: application/json" \
+                 -d '{
+                   "cloud_provider_region": "<your-cloud-deployment-region>",
+                   "cloud_provider_endpoint_id": "<your-vpc-endpoint-id>"
+                   "endpoint_type":"general"
```
[the remark this answers is not in the document]
Comment (krollins-mdb) [reply source/security/private-endpoints.txt:182]: Yep. That's valuable to expose here, I think.

File: source/security/private-endpoints.txt [hunk 10]
```diff
@@ -100,111 +93,193 @@ connection using either the App Services UI or the Admin API.
...
-            AWS CLI.
-
-            Modify the following command template with your private
-            endpoint information. Then, copy the command and run it in
-            your terminal:
-
+            a. In the App Services UI, navigate to :guilabel:`App Settings` and
+               then click the :guilabel:`Private Endpoints` tab.
+            #. Click :guilabel:`Add Private Endpoint`, then click
+               :guilabel:`General`.
```
[the remark this answers is not in the document]
Comment (krollins-mdb) [reply source/security/private-endpoints.txt:253]: Yep. I've removed the UI bits.

File: source/security/private-endpoints.txt (hunk 1 above)
Comment (krollins-mdb) [reply source/security/private-endpoints.txt:297]: Yeah, I think pointing them to the general endpoint section makes the most sense. I did that at first, then changed it to what you saw. Can't remember why.

And thanks for noticing the odd rendering. I fixed that for the general endpoint section, but forgot to update here. Should be good now!

File: source/security/private-endpoints.txt [hunk 11]
```diff
@@ -252,3 +325,49 @@ You can toggle this setting in three ways:
...
+Definitions
+-----------
+
+There are three kinds of Private Endpoints:
+
+- Legacy: Endpoints that use ``realm.mogodb.com``
+- General: TODO: Add definition
+- Sync: TODO: Add definition
+
+Supported SDK Versions
```
[the remark this answers is not in the document]
Comment (krollins-mdb) [reply source/security/private-endpoints.txt:337]: Excellent question. No, we don't need to repeat it. We can just link out to that section. Thank you for the reminder! We definitely don't want to keep it in two places.

**Comment by krollins-mdb** [comment]
@melodeeli98, thanks for your thorough review! I've updated based on your feedback. Do you want to take another look?

**Comment by melodeeli98** [comment]
@krollins-mdb yes, will take another look now!

**Review by melodeeli98 — APPROVED** [review]
A few remaining things, but otherwise LGTM! Thank you for doing this!

Reminder that we should hold off on publishing these changes until our upcoming release in two weeks :)

File: source/security/private-endpoints.txt [hunk 12]
```diff
@@ -63,33 +47,42 @@ following information:
...
-        [
-          {
-            "cloud_provider_region": "aws-us-east-1",
-            "name": "eps_baas-pl-prod_us-east-1_cloud",
-            "dns": "*.aws.services.cloud.mongodb.com",
-            "service_name": "com.amazonaws.vpce.us-east-1.vpce-svc-0f12ab34cd56ef789"
-          },
-          ...
-        ]
+After you've done everything in the Before You Begin section on this page,
```
Comment (cbullinger) [inline source/security/private-endpoints.txt:77]: so the private hosted zone _was_ a requirement?

File: source/security/private-endpoints.txt [hunk 13]
```diff
@@ -63,33 +47,42 @@ following information:
...
+you can add a new Private Endpoint. You can do this with the App Services
+UI or the Admin API.
 
-Once you have this information, you can set up the Private Endpoint
-connection using either the App Services UI or the Admin API.
+You can add a new General Endpoint or a new Sync Endpoint. You can also
```
Comment (cbullinger) [inline source/security/private-endpoints.txt:81]: (yeah, the definitions should come before we start talking about them like users already know what these mean)

File: source/security/private-endpoints.txt [hunk 14]
```diff
@@ -100,111 +93,82 @@ connection using either the App Services UI or the Admin API.
 
          .. step:: Create an Interface Endpoint in Your VPC
```
Comment (cbullinger) [inline source/security/private-endpoints.txt:94]: ... should I know what an Interface Endpoint is?

does it assume a specific endpoint type requirement? if it _has_ to be interface, why do i have to get it in the private endpoint svcs response (and the example output shows a "general" type)??

File: source/security/private-endpoints.txt [hunk 15]
```diff
@@ -100,111 +93,82 @@ connection using either the App Services UI or the Admin API.
...
-
-            #. Enter your **VPC ID** in the input box.
-
-            #. Enter your **Subnet IDs** as a comma-separated list in the input box.
-
-            Once you've entered this information, the fully constructed
-            AWS CLI command is displayed in the UI. Copy the command and
-            run it in your terminal. It should resemble the following:
-
+            a. In the App Services UI, navigate to :guilabel:`App Settings` and
```
Comment (cbullinger) [inline source/security/private-endpoints.txt:96]:
```suggestion
            a. In the App Services UI, navigate to :guilabel:`App Settings`,
```

File: source/security/private-endpoints.txt [hunk 16]
```diff
@@ -100,111 +93,82 @@ connection using either the App Services UI or the Admin API.
...
+               then click the :guilabel:`Private Endpoints` tab.
+            #. Click :guilabel:`Add Private Endpoint`, then click
+               :guilabel:`General`.
+            #. Add your VPC ID and Subnet IDs (comma-separated list), then click
```
Comment (cbullinger) [inline source/security/private-endpoints.txt:100]: are you putting all of these in one field in the updated UI? or are there still separate fields for vpc and subnet id entries? if so, maybe split it out somewhat?
```suggestion
            #. Add your VPC ID, your Subnet IDs (as a comma-separated list), then click
```

File: source/security/private-endpoints.txt [hunk 17]
```diff
@@ -100,111 +93,82 @@ connection using either the App Services UI or the Admin API.
...
+               :guilabel:`Next`.
+            #. In your terminal, create a VPC Endpoint with the command generated
```
Comment (cbullinger) [inline source/security/private-endpoints.txt:102]: feels like there's a step missing. when does the generated command display for me to copy? in the modal _after_ you click Next?

**Review by cbullinger — COMMENTED** [review]
I'm going to put this review down since I seem to have a lot of feedback; I'll probably pick it back up next week. 

There are some blocking comments that need to be addressed, though (like the tabbed procedures not displaying correctly)

File: source/security/private-endpoints.txt [hunk 18]
```diff
@@ -214,21 +178,136 @@ connection using either the App Services UI or the Admin API.
...
-            Use the AWS CLI to modify your VPC private DNS settings. This
-            ensures that your App can resolve the private endpoint's DNS name
-            to the correct IP address.
+Add a Sync Endpoint
+~~~~~~~~~~~~~~~~~~~
+
+Sync Endpoints require that you also have a General Endpoint associated with
+the Sync Endpoint.
+
+.. tabs-realm-admin-interfaces::
```
[the remark this answers is not in the document]
Comment (krollins-mdb) [reply source/security/private-endpoints.txt:193]: Huh. I can see them and the general endpoint section.

File: source/security/private-endpoints.txt [hunk 19]
```diff
@@ -63,33 +47,42 @@ following information:
...
+      curl https://services.cloud.mongodb.com/api/admin/v3.0/groups/{groupId}/apps/{appId}/security/private_endpoint_service_infos \
+        -H "Authorization: Bearer <access_token>"
+
+    .. output::
+      :language: json
+      :emphasize-lines: 6
 
-        curl https://services.cloud.mongodb.com/api/admin/v3.0/groups/{groupId}/apps/{appId}/security/private_endpoint_service_infos \
-          -H "Authorization: Bearer <access_token>"
+      [
```
[the remark this answers is not in the document]
Comment (krollins-mdb) [reply source/security/private-endpoints.txt:69]: Yep. Lindsey is working on that with this ticket: https://jira.mongodb.org/browse/DOCSP-38290

File: source/security/private-endpoints.txt (hunk 12 above)
Comment (krollins-mdb) [reply source/security/private-endpoints.txt:81]: Yeah, we should treat it as a requirement.

File: source/security/private-endpoints.txt
Comment (melodeeli98) [inline source/security/private-endpoints.txt:1]: one more thing actually - I think it would be worth noting somewhere that Private Link failures (e.g. request attempts outside of the VPC) will return a 404

File: source/security/private-endpoints.txt (hunk 18 above)
[the remark this answers is not in the document]
Comment (cbullinger) [reply source/security/private-endpoints.txt:197]: Yep, you can disregard this - I just checked and they're displaying for me now. I'll chalk it up to build gremlins or cache trolls.

File: source/security/private-endpoints.txt [hunk 20]
```diff
@@ -63,33 +47,42 @@ following information:
   you. If you're using the Admin API, you can find this value by
```
[the remark this answers is not in the document]
Comment (krollins-mdb) [reply source/security/private-endpoints.txt:51]: Thanks for pointing out the awkwardness. I've removed from this section and embedded in the API procedure, as it makes the most sense there.

File: source/security/private-endpoints.txt (hunk 14 above)
Comment (krollins-mdb) [reply source/security/private-endpoints.txt:98]: Yeah, good point. We don't need this to be so specific.

File: source/security/private-endpoints.txt (hunk 16 above)
Comment (krollins-mdb) [reply source/security/private-endpoints.txt:104]: I clarified this a bit.

File: source/security/private-endpoints.txt (hunk 14 above)
Comment (krollins-mdb) [reply source/security/private-endpoints.txt:98]: Actually, looking it over again, I do think we need to include "interface" here. I've simplified a bit to "Create a VPC Interface Endpoint".

The UI uses "VPC Interface Endpoint" and it's a helpful distinction to avoid confusion between the VPC endpoint and the Private Endpoint users are creating.

But maybe I didn't totally understand your question here.

File: source/security/private-endpoints.txt (hunk 17 above)
Comment (krollins-mdb) [reply source/security/private-endpoints.txt:106]: Yeah, you're right. The UI automatically generates it when you add the fields. You don't need to do anything. But I didn't explicitly say users should copy that generated endpoint. I've revised to make the process a bit clearer.

File: source/security/private-endpoints.txt [hunk 21]
```diff
@@ -98,113 +76,109 @@ connection using either the App Services UI or the Admin API.
...
-            Modify the following command template with your private
-            endpoint information. Then, copy the command and run it in
-            your terminal:
-
+         .. step:: Create a VPC Interface Endpoint
+
+            a. Call the :admin-api-endpoint:`List VPC Private Endpoint Services
+               <operation/adminListPrivateEndpointServiceInfo>` endpoint.
+            #. Filter the response object to find the entry for your App's
+               deployment region and the endpoint type you want to create.
```
Comment (dacharyc) [inline source/security/private-endpoints.txt:131]: Should we link to the [Cloud Deployment Regions](https://preview-mongodbkrollinsmdb.gatsbyjs.io/atlas-app-services/DOCSP-37831/apps/deployment-models-and-regions/#cloud-deployment-regions) documentation to make it easier for people to find their relevant deployment region?

Comment (dacharyc) [reply source/security/private-endpoints.txt:131]: And should we specify that the endpoint type here can be one of `general` or `sync`? (Assuming that's accurate.)

File: source/security/private-endpoints.txt [hunk 22]
```diff
@@ -98,113 +76,109 @@ connection using either the App Services UI or the Admin API.
...
+                     {
+                        "cloud_provider_region": "aws-us-east-1",
+                        "service_name": "com.amazonaws.vpce.us-east-1.vpce-svc-0718bec2c0bdeab5c",
+                        "endpoint_type": "general"
+                     },
+                     ...
+                     ]
+            #. Get the ``service_name`` field from the desired entry in the
+               previous step.
+            #. Create a VPC Endpoint using the AWS CLI `create-vpc-endpoint
```
Comment (dacharyc) [inline source/security/private-endpoints.txt:155]: Hmm, we're saying here to use the AWS CLI to create the endpoint, but this command looks exactly like the one in Step 1 of the UI procedure. Is that one _also_ supposed to be using the AWS CLI? If so, we should probably specify that there. (Around ln 98/99.)

Comment (dacharyc) [reply source/security/private-endpoints.txt:155]: Also, if we do expect users to use AWS CLI for _both_ versions of this procedure, should that be listed in the "Before You Begin" section - i.e. "Install and authenticate the AWS CLI"?

File: source/security/private-endpoints.txt [hunk 23]
```diff
@@ -214,21 +188,161 @@ connection using either the App Services UI or the Admin API.
                  -H "Content-Type: application/json" \
                  -d '{
                    "cloud_provider_region": "<your-cloud-deployment-region>",
-                   "cloud_provider_endpoint_id": "<your-vpc-endpoint-id>"
+                   "cloud_provider_endpoint_id": "<your-vpc-endpoint-id>",
+                   "endpoint_type": "general",
+                   "comment": "<comment-about-endpoint>"
```
Comment (dacharyc) [inline source/security/private-endpoints.txt:193]: Since users don't have the Admin API definition right here, should we make this placeholder more descriptive to make it easier for users to understand how to populate this? i.e.:
```suggestion
                   "comment": "Optional: add a comment describing the endpoint."
```

File: source/security/private-endpoints.txt [hunk 24]
```diff
@@ -214,21 +188,161 @@ connection using either the App Services UI or the Admin API.
...
 
-         .. step:: Modify Your VPC Private DNS
 
-            Use the AWS CLI to modify your VPC private DNS settings. This
-            ensures that your App can resolve the private endpoint's DNS name
-            to the correct IP address.
+Add a Sync Endpoint
+~~~~~~~~~~~~~~~~~~~
+
+Sync Endpoints require that you also have a General Endpoint associated with
```
Comment (dacharyc) [inline source/security/private-endpoints.txt:200]: I'm not quite sure I understand what this is trying to communicate. I wonder if we can reword this to be a little more explicit? If I'm interpreting this correctly, maybe something like:

> Every Sync Endpoint must be paired with a General Endpoint created using the same generated command.

File: source/security/private-endpoints.txt [hunk 25]
```diff
@@ -98,113 +76,109 @@ connection using either the App Services UI or the Admin API.
...
+            
+               .. code-block:: sh
 
+                 aws ec2 create-vpc-endpoint \
+                   --vpc-endpoint-type Interface
+                   --service-name <App Services Endpoint Service Name> \
+                   --vpc-id <your-vpc-id> \
+                   --region <your-aws-deployment-region> \
+                   --subnet-ids <your-subnet-ids>
+            #. In your terminal, create a Sync VPC Endpoint with the generated
```
Comment (dacharyc) [inline source/security/private-endpoints.txt:98]: I'm a little confused about the language we're using here, after reading the "Add a Sync Endpoint" section below. In ln 229, we say to create a Sync VPC Endpoint, and then we say to also create a General VPC Endpoint. So - should this one be a "General VPC Endpoint"? Or maybe a "VPC Endpoint"?

```suggestion
            #. In your terminal, create a General VPC Endpoint with the generated
```

File: source/security/private-endpoints.txt [hunk 26]
```diff
@@ -214,21 +188,161 @@ connection using either the App Services UI or the Admin API.
...
+            
+               .. code-block:: sh
+
+                 aws ec2 create-vpc-endpoint \
+                   --vpc-endpoint-type Interface
+                   --service-name <App Services Endpoint Service Name> \
+                   --vpc-id <your-vpc-id> \
+                   --region <your-aws-deployment-region> \
+                   --subnet-ids <your-subnet-ids>
+            #. In your terminal, create a Sync VPC Endpoint with the generated
```
Comment (dacharyc) [inline source/security/private-endpoints.txt:229]: Ok, definitely confused about this. Are we creating AWS "VPC Endpoints" here? If so, should we be calling this a "VPC Endpoint" instead of a "Sync VPC Endpoint"? Are we using "Sync" and "General" here to make it clearer the relationship between the VPC Endpoint and whether App Services is using it as a "General" or "Sync" endpoint?

If the latter, I think we should be crystal clear that these are AWS VPC Endpoints that you will then add to App Services as either a "General" or "Sync" endpoint. And if I am totally misunderstanding this, maybe we can otherwise clarify the language? (Or feel free to disregard if this should be obvious to people who are more familiar with Private Endpoints.)

File: source/security/private-endpoints.txt [hunk 27]
```diff
@@ -214,21 +188,161 @@ connection using either the App Services UI or the Admin API.
...
+      :tabid: api
+
+      .. procedure::
+
+         .. step:: Create a VPC Interface Endpoint
+
+            a. Call the :admin-api-endpoint:`List VPC Private Endpoint Services
+               <operation/adminListPrivateEndpointServiceInfo>` endpoint.
+            #. Filter the response object to find the entry for your App's
+               deployment region and the endpoint type you want to create.
```
Comment (dacharyc) [inline source/security/private-endpoints.txt:265]: Same Q as above - should we link out to the Cloud Deployment Regions docs here to make it easier for users to figure out the appropriate region, and/or list that endpoint types can be one of `general` and `sync`?

File: source/security/private-endpoints.txt [hunk 28]
```diff
@@ -214,21 +188,161 @@ connection using either the App Services UI or the Admin API.
...
+
+                  curl https://services.cloud.mongodb.com/api/admin/v3.0/groups/{groupId}/apps/{appId}/security/private_endpoints \
+                  -X POST \
+                  -H "Authorization: Bearer <accessToken>" \
+                  -H "Content-Type: application/json" \
+                  -d '{
+                     "cloud_provider_region": "<your-cloud-deployment-region>",
+                     "cloud_provider_endpoint_id": "<your-vpc-endpoint-id>",
+                     "endpoint_type": "sync",
+                     "comment": "<comment-about-endpoint>"
```
Comment (dacharyc) [inline source/security/private-endpoints.txt:328]: Same Q as above about adding a more descriptive comment here since we're divorced from the Admin API definition.

File: source/security/private-endpoints.txt [hunk 29]
```diff
@@ -214,21 +188,161 @@ connection using either the App Services UI or the Admin API.
...
+                   --subnet-ids <your-subnet-ids>
+            #. In your terminal, create a Sync VPC Endpoint with the generated
+               command.
+            #. You must also create a General VPC Endpoint using the command
+               App Services generates for the General Endpoint.
+         
+         .. step:: Add the Private Endpoints to Your App
+
+            Now that you've created the interface endpoint in your VPC, you
+            need to add it to your App.
```
Comment (dacharyc) [inline source/security/private-endpoints.txt:237]: Are we creating more than one VPC endpoint in the step above - i.e. one for Sync and one for General? If so, we should probably match the plural here - i.e.

```suggestion
            Now that you've created the interface endpoints in your VPC, you
            need to add them to your App. 
```

File: source/security/private-endpoints.txt [hunk 30]
```diff
@@ -214,21 +188,161 @@ connection using either the App Services UI or the Admin API.
...
+               respective ``Your VPC Endpoint ID`` field in the App Services UI,
+               then click :guilabel:`Next`.
+            #. If this is the first Private Endpoint you are creating, you will 
+               see a toggle you can use to either restrict or open all access from 
+               requests outside your PrivateLink connection. By default, App 
+               Services blocks all requests coming from outside your PrivateLink 
+               connection. This setting applies to this endpoint and **all 
+               additional endpoints you create**. Refer to :ref:`change access 
+               restrictions <change_access_restrictions>` for more information.
+            #. Click :guilabel:`Finish Endpoint Creation`.
```
Comment (dacharyc) [inline source/security/private-endpoints.txt:253]: In the Admin API procedure, we explicitly call out as a separate step:

> You must also create a General Private Endpoint by following the steps for creating a General Private Endpoint on this page.

Should we do the same thing here in the UI procedure?

File: source/security/private-endpoints.txt [hunk 31]
```diff
@@ -214,21 +188,161 @@ connection using either the App Services UI or the Admin API.
...
+Migrate a Legacy Endpoint
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+If you have an endpoint that uses the legacy ``realm.mongodb.com`` domain, you
+must migrate it to use the new ``services.cloud.mongodb.com`` domain. You can't
+create new legacy endpoints.
+
+**To migrate a Legacy Endpoint:** 
+
+#. Create new endpoints to replace existing VPC endpoints
```
Comment (dacharyc) [inline source/security/private-endpoints.txt:342]: For this step, do we mean users must go through the entire process to create new General or Sync endpoints - i.e. create new VPC Interface Endpoints and add them to App Services as a General or Sync Private Endpoint? If so, I wonder if we want to be more explicit? Maybe something like:

> Following the procedures on this page, create new General or Sync VPC Endpoints and add them to your App Services App.

File: source/security/private-endpoints.txt [hunk 32]
```diff
@@ -252,3 +366,35 @@ You can toggle this setting in three ways:
...
+- You can't use Private Endpoints with globally deployed Apps or local
+  Apps deployed to Azure or GCP.
+
+- You can't use public internet connections to access your App if you
+  have a Private Endpoint enabled. All requests must come through the
+  Private Endpoint.
+
+- If you :ref:`change your deployment model <change-deployment-models>`,
+  then your existing Private Endpoints will not continue to work. You
+  will have to create new Private Endpoints for the App's new region.
```
Comment (dacharyc) [inline source/security/private-endpoints.txt:389]: Nit: our style guide calls for us to avoid future tense. Maybe we can simplify this with something like: 

```suggestion
  then you must create new Private Endpoints for the App's new region.
```

**Review by dacharyc — APPROVED** [review]
I have a lot of nits and suggestions, but in the interest of timing, I'd say none of them is blocking. Feel free to pick and choose which ones seem worth applying given the time constraints.

File: source/security/private-endpoints.txt (hunk 25 above)
Comment (krollins-mdb) [reply source/security/private-endpoints.txt:98]: Yep. That should definitely be "General".

File: source/security/private-endpoints.txt [hunk 33]
```diff
@@ -4,92 +4,70 @@
...
-  response.
-
-  .. io-code-block::
+Set up a private hosted zone for your VPC. You need your VPC ID for this.
+Use `ModifyVpcAttribute <https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_ModifyVpcAttribute.html>`__
+to set the following VPC attributes to true:
 
-      .. input::
-        :language: sh
+- ``enableDnsHostnames``
```
[the remark this answers is not in the document]
Comment (krollins-mdb) [reply source/security/private-endpoints.txt:56]: Hmm. Yeah, looks like they should be capitalized.

File: source/security/private-endpoints.txt (hunk 21 above)
Comment (krollins-mdb) [reply source/security/private-endpoints.txt:129]: I don't think we need to specify that it can be sync or general. In the "Add a Sync Endpoint" section, it's changed to sync.

File: source/security/private-endpoints.txt (hunk 22 above)
Comment (krollins-mdb) [reply source/security/private-endpoints.txt:153]: Yep. Good catch. It should absolutely be in the Before You Begin section.

File: source/security/private-endpoints.txt (hunk 26 above)
Comment (krollins-mdb) [reply source/security/private-endpoints.txt:227]: Yeah, the language here isn't clear. It should be just VPC Endpoint. I think that's clear enough.

File: source/security/private-endpoints.txt [hunk 34]
```diff
@@ -4,207 +4,182 @@
...
+Gather the following information:
 
 - Your **VPC ID**, which you can find in the AWS VPC dashboard or
-  by running the :aws:`describe-vpcs
-  </cli/latest/reference/ec2/describe-vpcs>` command in the AWS
-  CLI.
-
+by running the :aws:`describe-vpcs
+</cli/latest/reference/ec2/describe-vpcs>` command in the AWS
+CLI.
```
Comment (cbullinger) [inline source/security/private-endpoints.txt:46]:
```suggestion
  by running the :aws:`describe-vpcs
  </cli/latest/reference/ec2/describe-vpcs>` command in the AWS
  CLI.
```

File: source/security/private-endpoints.txt [hunk 35]
```diff
@@ -4,207 +4,182 @@
...
-            "service_name": "com.amazonaws.vpce.us-east-1.vpce-svc-0f12ab34cd56ef789"
-          },
-          ...
-        ]
-
-Once you have this information, you can set up the Private Endpoint
-connection using either the App Services UI or the Admin API.
+or by running the :aws:`describe-subnets
+</cli/latest/reference/ec2/describe-subnets>` command in the AWS
+CLI.
```
Comment (cbullinger) [inline source/security/private-endpoints.txt:50]:
```suggestion
  or by running the :aws:`describe-subnets
  </cli/latest/reference/ec2/describe-subnets>` command in the AWS
  CLI.
  
```
